### PR TITLE
Feat/add fela hook

### PR DESCRIPTION
**Pull request by horaklukas**
Add `useFelaEnhanced` hook and refactor components to use it.

### CODE REVIEW

File: packages/cra-template-typescript/template/src/hooks/useFelaEnhanced.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { combineMultiRules } from 'fela-tools';
+import { useFela } from 'react-fela';
+
+// This solution is based on the implementation of [connect binding](https://github.com/robinweser/fela/blob/master/packages/fela-bindings/src/connectFactory.js) from fela
+export default function useFelaEnhanced(rules, { extend, ...otherProps } = {}) {
```
Comment (jstorm31) [inline packages/cra-template-typescript/template/src/hooks/useFelaEnhanced.ts]: It'd be nice if at least function parameters had typings. Ideally from fela package, but I added [custom types on rafflu](https://gitlab.ack.ee/Web/rafflu/-/blob/development/src/hooks/useFelaEnhanced.ts) to set it up fast.

**Review by Tomas2D — APPROVED** [review]

File: packages/cra-template-typescript/template/src/hooks/useFelaEnhanced.ts (hunk 1 above)
Comment (horaklukas) [reply packages/cra-template-typescript/template/src/hooks/useFelaEnhanced.ts]: Done ✅

File: packages/cra-template-typescript/template/src/hooks/useFelaEnhanced.ts [hunk 2]
```diff
@@ -1,18 +1,34 @@
 import React from 'react';
-import { combineMultiRules } from 'fela-tools';
-import { useFela } from 'react-fela';
+import { combineMultiRules, TNormalizedMultiRule } from 'fela-tools';
+import { TRule } from 'fela';
+import { useFela, Rules as ReactFelaRules } from 'react-fela';
+
+import { Theme } from 'styles/theme';
```
Comment (jstorm31) [inline packages/cra-template-typescript/template/src/hooks/useFelaEnhanced.ts]: Is `Theme` type really exported from there?

File: packages/cra-template-typescript/template/src/hooks/useFelaEnhanced.ts (hunk 1 above)
Comment (jstorm31) [reply packages/cra-template-typescript/template/src/hooks/useFelaEnhanced.ts]: Great 👍

File: packages/cra-template-typescript/template/src/hooks/useFelaEnhanced.ts (hunk 2 above)
Comment (horaklukas) [reply packages/cra-template-typescript/template/src/hooks/useFelaEnhanced.ts]: good point, will check

Comment (horaklukas) [reply packages/cra-template-typescript/template/src/hooks/useFelaEnhanced.ts]: Add `Theme` type

**Review by jstorm31 — APPROVED** [review]

**Review by cermakjiri — APPROVED** [review]

**Pull request merged**